### PR TITLE
feat: llama index basic data loader

### PR DESCRIPTION
**Pull request by scottschreckengaust**
This is the initial sample repository for running the LlamaIndex data loader construct.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: samples/llamaindex-basic-data-loader/README.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+![architecture diagram](docs/llamaindex-basic-data-loader.png)
+
+A couple resources are also noted:
+* The "circuit breaker" Amazon System Manager Parameter can 'pause' processing the queue
+* The "container" Amazon ECR image that runs the code and loads the requirements for the reader
+
+## Setup
+
+This project uses the CDK with Python
```
Comment (krokoko) [inline samples/llamaindex-basic-data-loader/README.md:36]: We can use the same structure here for the pre-requisites as we do for other samples, for instance: https://github.com/aws-samples/generative-ai-cdk-constructs-samples/tree/main/samples/sagemaker_huggingface_model_svd#prerequisites

File: samples/llamaindex-basic-data-loader/app.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env python3
```
Comment (krokoko) [inline samples/llamaindex-basic-data-loader/app.py:1]: header license missing

File: samples/llamaindex-basic-data-loader/app.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+import os
+
+import aws_cdk as cdk
```
Comment (krokoko) [inline samples/llamaindex-basic-data-loader/app.py:4]: Missing cdk-nag enabled

File: samples/llamaindex-basic-data-loader/bin/llamaindex_basic_data_loader_stack.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
+from os import (
```
Comment (krokoko) [inline samples/llamaindex-basic-data-loader/bin/llamaindex_basic_data_loader_stack.py:1]: missing license header file

File: samples/llamaindex-basic-data-loader/docker/pid [hunk 5]
```diff
@@ -0,0 +1 @@
+5978
```
Comment (krokoko) [inline samples/llamaindex-basic-data-loader/docker/pid:1]: why is this value harcoded ?

File: samples/llamaindex-basic-data-loader/docker/sqs_consumer.py [hunk 6]
```diff
@@ -0,0 +1,252 @@
+#!/bin/env python3 -x -u
```
Comment (krokoko) [inline samples/llamaindex-basic-data-loader/docker/sqs_consumer.py:1]: Missing license header file

File: samples/llamaindex-basic-data-loader/tests/unit/test_llamaindex_basic_data_loader_stack.py [hunk 7]
```diff
@@ -0,0 +1,38 @@
+import aws_cdk as core
```
Comment (krokoko) [inline samples/llamaindex-basic-data-loader/tests/unit/test_llamaindex_basic_data_loader_stack.py:1]: Missing license header file

**Review by krokoko — CHANGES_REQUESTED** [review]
Mostly missing license header files and cdk-nag

**Comment by krokoko** [comment]
Also please add the sample to the samples catalog in the main readme :) https://github.com/aws-samples/generative-ai-cdk-constructs-samples/tree/main?tab=readme-ov-file#getting-started

File: samples/llamaindex-basic-data-loader/docker/pid (hunk 5 above)
Comment (scottschreckengaust) [reply samples/llamaindex-basic-data-loader/docker/pid:1]: committed unintentionally

File: samples/llamaindex-basic-data-loader/app.py (hunk 2 above)
Comment (scottschreckengaust) [reply samples/llamaindex-basic-data-loader/app.py:1]: added

File: samples/llamaindex-basic-data-loader/app.py (hunk 3 above)
Comment (scottschreckengaust) [reply samples/llamaindex-basic-data-loader/app.py:15]: included cdk_nag

File: samples/llamaindex-basic-data-loader/bin/llamaindex_basic_data_loader_stack.py (hunk 4 above)
Comment (scottschreckengaust) [reply samples/llamaindex-basic-data-loader/bin/llamaindex_basic_data_loader_stack.py:13]: added

File: samples/llamaindex-basic-data-loader/docker/sqs_consumer.py (hunk 6 above)
Comment (scottschreckengaust) [reply samples/llamaindex-basic-data-loader/docker/sqs_consumer.py:1]: added

File: samples/llamaindex-basic-data-loader/tests/unit/test_llamaindex_basic_data_loader_stack.py (hunk 7 above)
Comment (scottschreckengaust) [reply samples/llamaindex-basic-data-loader/tests/unit/test_llamaindex_basic_data_loader_stack.py:13]: done

**Comment by scottschreckengaust** [comment]
> Also please add the sample to the samples catalog in the main readme :) https://github.com/aws-samples/generative-ai-cdk-constructs-samples/tree/main?tab=readme-ov-file#getting-started

done

**Comment by scottschreckengaust** [comment]
> Mostly missing license header files and cdk-nag

added, and done.

**Comment by scottschreckengaust** [comment]
> Also please add the sample to the samples catalog in the main readme :) https://github.com/aws-samples/generative-ai-cdk-constructs-samples/tree/main?tab=readme-ov-file#getting-started

added

**Review by krokoko — APPROVED** [review]

**Pull request merged**